test/compareLocations.test.js

```diff
@@ -0,0 +1,152 @@
+var should = require("should");
+var compareLocations = require("../lib/compareLocations");
+var createLocation = function(overides) {
```

Comment:
In fact, inside the `compareLocations` functions, it has some argument reassignments. It should be refactored for performance and clearance